easydigitaldownloads/edd-discounts-pro
# Release/1.5

Just needs changelog and then it's good to deploy.

## What this PR comprises of

This PR comprises the 6 tickets of the 1.5 major release of Discounts Pro: https://github.com/chriscct7/edd-discounts/issues?utf8=%E2%9C%93&q=is%3Aissue%20milestone%3A1.5

Futhermore it solves the following 10 unticketed items:
- The date fields in the admin used untranslatable strings
- Significant reduction in the amount of duplicate/unncessary code
- Simplification and betterment of save routine
- Significantly better HTML structure of admin settings page
- Removal of nearly all styling. Only a couple CSS rules left, now done inline to remove entire assets folder, and reduce pageload.
- Admin page code has been significantly cleaned up.
- Renamed EDD_Admin to EDD_DP_Admin to reduce chance of future issue
- Rewrote how the values on the discounts list table were done. Should make it much more accurate + informative.
- Removes unused product class
- Removes default.po
- Net removal of nearly **_5,000_** lines of code :smile: 
## How to test release and what to test

Please test using variations in the admin settings panel as this is the primary bug we're fixing. To test, please make sure to use https://github.com/easydigitaldownloads/easy-digital-downloads/pull/4895 as your EDD version. Note, you will need to alter the EDD version of that branch to say 2.6.8 in order to activate Discounts Pro 1.5
## Summary of Release

Despite being a major release by version number, this release is really a huge maintenance release, seeking to reduce support by fixing the most major outstanding bugs and fixes most outstanding bugs by revamping the admin panel to use the EDD core HTML class and fixes issues with dates/times. It does not introduce any significant new functionality (though all select fields can now use Chosen in the admin). This release prepares Discounts Pro for the future by cleaning up most of the admin side of the plugin so we can start getting ready to add significant new functionality to it.
